plc4j/drivers/opcua/src/test/java/org/apache/plc4x/java/opcua/OpcuaPlcDriverTest.java

```diff
@@ -516,21 +519,24 @@ public void run() {
                         PlcReadResponse read_response = read_request.execute().get();
                         assertThat(read_response.getResponseCode("Bool")).isEqualTo(PlcResponseCode.OK);
                     }
-
+                    this.latch.countDown();
                 } catch (ExecutionException e) {
                     LOGGER.error("run aborted", e);
+                    this.latch.countDown();
                 } catch (InterruptedException e) {
-                    Thread.currentThread().interrupt();
```

Comment:
this must stay in

Reply:
because this ensures that the interrupt flag stays set https://github.com/joansmith/sonar-java/blob/master/java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S2142.html

Reply:
Out of curiosity: why did you remove the line?

Reply:
Only one case for interrupted exception to happen is when test is terminated before completing 100 read or write attempts. Because exception is caught outside of the loop thread is effectively terminated at the time exception is caught.